### PR TITLE
Get samples list from remote repository

### PR DESCRIPTION
**Pull request by thorsten-stripe**
Closes #393 

 ### Reviewers
r? @tomer-stripe 
cc @stripe/dev-platform

 ### Summary
* Use the existing clone, cache, and pull mechanisms to get the samples list from a remote repository: https://github.com/stripe-samples/samples-list

@tomer-stripe could you review? Not a Golang expert and I'm not sure it's a robust implementation. Also `samples_test.go` is failing :(

### CODE REVIEW

File: pkg/cmd/samples/create.go [hunk 1]
```diff
@@ -35,10 +35,9 @@ func NewCreateCmd(config *config.Config) *CreateCmd {
 		forceRefresh: false,
 	}
 	createCmd.Cmd = &cobra.Command{
-		Use:       "create <sample> [destination]",
-		ValidArgs: samples.Names(),
```
Comment (tomer-stripe) [inline pkg/cmd/samples/create.go]: Just occurred to me: one downside of this approach is that the names will no longer autocomplete. Not a blocker but calling it out.

File: pkg/samples/list.go [hunk 2]
```diff
@@ -24,172 +35,94 @@ func (sd *SampleData) GitRepo() string {
...
+		return err
+	}
 
-var oxxoPayment = &SampleData{
-	Name:        "oxxo-payment",
-	Description: "Build a payment form to collect OXXO and card payments",
-	URL:         "https://github.com/stripe-samples/oxxo-payment",
-}
+	if _, err := s.Fs.Stat(listPath); os.IsNotExist(err) {
+		err = s.Git.Clone(listPath, "https://github.com/stripe-samples/samples-list.git")
```
Comment (tomer-stripe) [inline pkg/samples/list.go]: Could you set the url as a `const`?

File: pkg/samples/list.go [hunk 3]
```diff
@@ -24,172 +35,94 @@ func (sd *SampleData) GitRepo() string {
...
 
-var connectOnboardingStandard = &SampleData{
-	Name:        "connect-onboarding-for-standard",
-	Description: "Learn how to onboard users for Stripe Connect with Onboarding for Standard",
-	URL:         "https://github.com/stripe-samples/connect-onboarding-for-standard",
-}
+// GetSamples returns a list that contains a mapping of Stripe Samples that
+// we want to be available in the CLI to some of their metadata.
+// TODO: what do we want to name these for it to be easier for users to select?
+// TODO: should we group them by products for easier exploring?
```
Comment (tomer-stripe) [inline pkg/samples/list.go]: @adreyfus-stripe do you have any thoughts here?

Comment (adreyfus-stripe) [reply pkg/samples/list.go]: being able to filter by product could be neat!

File: pkg/cmd/samples/create.go (hunk 1 above)
Comment (thorsten-stripe) [reply pkg/cmd/samples/create.go]: Is this autocomplete via tab key? That doesn't seem to work today either.

File: pkg/samples/list.go (hunk 3 above)
Comment (thorsten-stripe) [reply pkg/samples/list.go]: I'll leave that for a separate sample. Discussing with @ctrudeau-stripe how to best approach this.

**Comment by thorsten-stripe** [comment]
Thanks for the review! I've moved the URL to a const and fixed the test. 

PTAL :)

File: pkg/cmd/samples/create.go (hunk 1 above)
Comment (tomer-stripe) [reply pkg/cmd/samples/create.go]: Yea, I think the homebrew installation stuff is busted :(. Probably not a big deal though, this doesn't seem like something people are going to tab complete.

**Pull request merged**